cairo/ethereum/cancun/trie.cairo

```diff
@@ -0,0 +1,314 @@
+//     // elif isinstance(node, Bytes):
+//     // return node
+//     // else:
+//     // return previous_trie.encode_node(node, storage_root)
+//         // assert storage_root is not None
+//         // return encode_account(node, storage_root)
+//     // else:
+//         // if isinstance(node, (LegacyTransaction, Receipt, Withdrawal, U256)):
+//         // return encode(node)
+//         // elif isinstance(node, Bytes):
+//         // return node
+//         // else:
+//         // return previous_trie.encode_node(node, storage_root)
+//             // return encode(node)
+//         // else:
+//             // if isinstance(node, Bytes):
+//             // return node
+//             // else:
+//             // return previous_trie.encode_node(node, storage_root)
+//                 // return node
+//             // else:
+//                 // return previous_trie.encode_node(node, storage_root)
+// }
+
+// func copy_trie(trie: Trie[K, V]) -> Trie[K, V] {
+//     // Implementation:
+//     // return Trie(trie.secured, trie.default, copy.copy(trie._data))
+// }
+
+// func trie_set(trie: Trie[K, V], key: K, value: V) {
+//     // Implementation:
+//     // if value == trie.default:
+//     // if key in trie._data:
+//     // del trie._data[key]
+//     // else:
+//     // trie._data[key] = value
+//         // if key in trie._data:
+//         // del trie._data[key]
+//             // del trie._data[key]
+//     // else:
+//         // trie._data[key] = value
+// }
+
+// func trie_get(trie: Trie[K, V], key: K) -> V {
+//     // Implementation:
+//     // return trie._data.get(key, trie.default)
+// }
+
+// func common_prefix_length(a: Sequence, b: Sequence) -> felt {
+//     // Implementation:
+//     // for i in range(len(a)):
+//     // if i >= len(b) or a[i] != b[i]:
+//     // return i
+//         // if i >= len(b) or a[i] != b[i]:
+//         // return i
+//             // return i
+//     // return len(a)
+// }
+
+// func nibble_list_to_compact(x: Bytes, is_leaf: bool) -> Bytes {
+//     // Implementation:
+//     // compact = bytearray()
+//     // if len(x) % 2 == 0:
+//     // compact.append(16 * (2 * is_leaf))
+//     // for i in range(0, len(x), 2):
+//     // compact.append(16 * x[i] + x[i + 1])
+//     // else:
+//     // compact.append(16 * (2 * is_leaf + 1) + x[0])
+//     // for i in range(1, len(x), 2):
+//     // compact.append(16 * x[i] + x[i + 1])
+//         // compact.append(16 * (2 * is_leaf))
+//         // for i in range(0, len(x), 2):
+//         // compact.append(16 * x[i] + x[i + 1])
+//             // compact.append(16 * x[i] + x[i + 1])
+//     // else:
+//         // compact.append(16 * (2 * is_leaf + 1) + x[0])
+//         // for i in range(1, len(x), 2):
+//         // compact.append(16 * x[i] + x[i + 1])
+//             // compact.append(16 * x[i] + x[i + 1])
+//     // return Bytes(compact)
+// }
+
+func bytes_to_nibble_list{bitwise_ptr: BitwiseBuiltin*}(bytes_: Bytes) -> Bytes {
```

Comment:
1 bitwise = 4 range check, price wise. so i think it's equivalent (nothing to do, just noting)